src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Networking/INetworkConnection.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    /// <summary>
+    /// Classes that implement this interface represent network connections and allow sending messages across a network.
+    /// </summary>
+    public interface INetworkConnection
+    {
+        /// <summary>
+        /// IP address for the connected peer
+        /// </summary>
+        string Address { get; }
+
+        /// <summary>
+        /// Returns true if the network connection is connected, otherwise false
+        /// </summary>
+        bool IsConnected { get; }
+
+        /// <summary>
+        /// Call to set the network connection state to disconnected
+        /// </summary>
+        void Disconnect();
+
+        /// <summary>
+        /// Checks whether the associated client is still active. If not, the client is disconnected.
+        /// </summary>
+        /// <param name="currentTime">Time to use relative to last active timestamp to determine whether to disconnect</param>
+        void CheckConnectionTimeout(DateTime currentTime);
+
+        /// <summary>
+        /// Call to start enqueuing incoming messages
+        /// </summary>
+        /// <param name="incomingQueue">Queue used for enqueuing messages</param>
+        void QueueIncomingMessages(ConcurrentQueue<IncomingMessage> incomingQueue);
+
+        /// <summary>
+        /// Call to stop enqueuing incoming messages
+        /// </summary>
+        void StopIncomingMessageQueue();
```

Comment:
Moving this to UnsubscribeToIncomingMessages. If folks would pref this to be a TryUnsubscribeToIncomingMessages I can update implementation, but I think this should be viewed as a subscription/unsubscription to messages, or at least that is how it is used by TCPConnectionManager.

Reply:
Suggestion: SetMessageQueue(null) (see the comment above) removes this method.

Reply:
SetMessageQueue(null) doesn't have any safe guard to make sure that you aren't unsetting someone else's message queue. In this context, we only have we only have one setter, but this interface is public so folks may try and use it in a different manner than we do currently. I'd prefer having this as TrySubscribe/Unsubscribe so that we don't rule out multiple listeners. I'd also prefer not to rely SetMessageQueue(null) as a way of unregistering in case multiple users interact with this interface.